### PR TITLE
Sam3 cpuid

### PR DESCRIPTION
**Pull request by nqd**
Adding cpu id for sam3 cpu, inspiring from https://github.com/RIOT-OS/RIOT/issues/1511.

### CODE REVIEW

File: cpu/sam3/periph/cpuid.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+/**
+ * @addtogroup  driver_periph
+ * @{
+ *
+ * @file        cpuid.c
+ * @brief       Low-level CPUID driver implementation
+ *
+ * @author      Dinh Nguyen <nqdinhddt@gmail.com>
+ */
```
Comment (haukepetersen) [inline cpu/sam3/periph/cpuid.c]: you should add a closing bracket for the doxygen here

File: cpu/sam3/periph/cpuid.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    - The Unique Identifier is located in the first 128 bits of the Flash memory mapping. So, at the address
+    0x400000-0x40008F.
+    - To stop the Unique Identifier mode, the user needs to send the Stop Read unique Identifier command (SPUI)
+    by writing the Flash Command Register with the SPUI command.
+    - When the Stop read Unique Identifier command (SPUI) has been performed, the FRDY bit in the Flash
+    Programming Status Register (EEFC_FSR) rises. If an interrupt was enabled by setting the FRDY bit in
+    EEFC_FMR, the interrupt line of the NVIC is activated.
+    */
+
+    EFC1->EEFC_FCR = (0x5A << 24) | EFC_FCMD_STUI;
```
Comment (haukepetersen) [inline cpu/sam3/periph/cpuid.c]:
```c
#define EFC_KEY            0x5a
...
EFC1->EEFC_FCR = (EEFC_FCR_FKEY(EFC_KEY) | EFC_FCMD_STUI);
```
might be slightly nicer to read and would get rid of any magic numbers here...

File: cpu/sam3/periph/cpuid.c [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    uint32_t status;
+    do
+    {
+        status = EFC1->EEFC_FSR ;
+    } while ( (status & EEFC_FSR_FRDY) == EEFC_FSR_FRDY ) ;
```
Comment (haukepetersen) [inline cpu/sam3/periph/cpuid.c]: simplify to 
```c
while (EFC1->EEFC_FSR & EEFC_FSR_FRDY);
```

**Comment by haukepetersen** [comment]
Looking good, just some minor style related things. I don't have board here so I will only be able to test this beginning of next week...

**Comment by nqd** [comment]
thank for commenting @haukepetersen. I cleanup the code little bit with your suggestion.

**Comment by haukepetersen** [comment]
Gode looks good and seems to work (at least I get two different IDs on different arduino-due boards).

Could you please squash your commits? The best woud be to have one commit for the actual driver, and one for each board. The convention for RIOT is to start your commit with the module you change, e.g. `cpu/sam3: added cpuid periph driver implementation` or `board/arduino-due: enabled periph_cpuid feature` or something like this.

**Comment by nqd** [comment]
Nice idea for squashing.
I updated with 3 commits. Could you please check, and merge if possible :).

**Comment by haukepetersen** [comment]
could you just rename the cpu commit to something starting with `cpu/sam3: ` to comply to RIOT conventions? Thx

**Comment by nqd** [comment]
Great thanks @haukepetersen; git rebase to make commit follow your suggestion. Do the commits look good for RIOT conventions now?

**Comment by haukepetersen** [comment]
yes, ACK. Will merge once Travis is green...

**Pull request merged**